pennylane/ops/op_math/decompositions/controlled_decompositions.py

```diff
@@ -768,6 +768,44 @@ def _decompose_mcx_with_no_worker(wires, **_):
 
 decompose_mcx_with_no_worker = flip_zero_control(_decompose_mcx_with_no_worker)
 
+
+def _decompose_mcx_temporary_and_resource(num_control_wires, **__):
+    return {
+        qml.TemporaryAND: num_control_wires - 1,
+        adjoint_resource_rep(qml.TemporaryAND, {}): num_control_wires - 1,
+        qml.CNOT: 1,
+    }
+
+
+def condition(num_control_wires, num_work_wires, **_):
+    """Auxiliar condition used in the TemporaryAnd decomposition"""
+    if num_work_wires > num_control_wires - 1:
+        return True
+    return False
+
+
+@register_condition(condition)
```

Comment:
I think it should be a private function?
And `num_control_wires-1` work wires are sufficient, right?
```suggestion
def _condition_elbow_decomp(num_control_wires, num_work_wires, **_):
    """Auxiliary condition on the number of control wires, used in the ``TemporaryAnd`` decomposition"""
    if num_work_wires >= num_control_wires - 1:
        return True
    return False


@register_condition(_condition_elbow_decomp)
```